tests/test__mongomock.py

```diff
@@ -768,6 +768,23 @@ def test__unset(self):
         self.cmp.do.update({'name': 'bob'}, {'$unset': {'a': False}})
         self.cmp.compare.find({'name': 'bob'})
 
+    def test__unset_nested(self):
+        self.cmp.do.update({'_id': 1}, {'$set': {'a': {'b': 1, 'c': 2}}}, upsert=True)
+        self.cmp.do.update({'_id': 1}, {'$unset': {'a.b': True}})
+        self.cmp.compare.find()
+
+        self.cmp.do.update({'_id': 1}, {'$set': {'a': {'b': 1, 'c': 2}}}, upsert=True)
+        self.cmp.do.update({'_id': 1}, {'$unset': {'a.b': False}})
```

Comment:
You're trying to check here if the value of a field specifier does not impact the unset-operation, right?